### ISSUE TITLE
[V3] Checkbox Label in Code Behind C#

### ISSUE THREAD

**Issue by Fatihpk** [UXDivers/Grial-UI-Kit-Support#589]
I am using your V3 of Grial Kit and we are facing a problem with your checkboxes. 

We want to create the Checkbox on runtime from the Code behind and we CAN create the checkbox in Runtime. But we can't add the label on it:

**This works in XAML:**
`<grial:Checkbox IsChecked="true" HorizontalOptions="Start">
   <Label Margin="8,0,0,0" Text="Grial Checkbox Control" VerticalOptions="Center" />
</grial:Checkbox>`

We want use the same thing but only in code behind:

**We can't add the Label in Runtime:**
`var checkboxP = new UXDivers.Grial.Checkbox { IsChecked = false, HorizontalOptions = LayoutOptions.Start };`

We try this but does not work also:
_Create the Checkbox Label_
`var lblCheckBox = new Label { Margin = new Thickness(8, 0, 0, 0), Text = "Grial Checkbox Control", VerticalOptions = LayoutOptions.Center };`

_Adding the label to the Checkbox does not support it also:_
`checkboxP.Children.Add(lblCheckBox, 0, 0);`

**Comment by mrutter** [comment]
Hi @Fatihpk,
try this, it should work:

`            var checkboxP = new UXDivers.Grial.Checkbox {
                IsChecked = false,
                HorizontalOptions = LayoutOptions.Start,
                Content = new Label { Margin = new Thickness(8, 0, 0, 0), Text = "Grial Checkbox Control from Code", VerticalOptions = LayoutOptions.Center }
            };
`

Bye.
-m

**Comment by Fatihpk** [comment]
Hi @mrutter , thank you. 

How do you do this in c#?

**Fontsize:** 
`FontSize="{ grial:OnOrientationDouble PortraitPhone=16, LandscapePhone=16, PortraitTablet=18, LandscapeTablet=18 }"`

And also this:

**ApplyIOSSafeAreaAsPadding**
`grial:Effects.ApplyIOSSafeAreaAsPadding="Left,Right"`

**Comment by mrutter** [comment]
Hi @Fatihpk,
there are many ways to accomplish with your code.

**FontSize** This requires a binding because of changes of the value based on orientation/idiom. A solution should be the following:
`            var myLabel = new Label();
            var ooDouble = new OnOrientationDouble { PortraitPhone = 16, LandscapePhone = 16, PortraitTablet = 18, LandscapeTablet = 18 };
            myLabel.SetBinding(Label.FontSizeProperty, (ooDouble.ProvideValue(null) as BindingBase));
`

**ApplyIOSSafeAreaAsPadding** and **Picker BorderStyle** are both Attached Properties. The code looks like this:
`            var myPicker = new Picker();
            PickerProperties.SetBorderStyle(myPicker, BorderStyle.Rect);
`

`            var myView = new ContentView();
            UXDivers.Grial.Effects.SetApplyIOSSafeAreaAsPadding(myView, IOSSafeArea.Left | IOSSafeArea.Right);
`

**Label Icon** Is a simple static value thus your code will be:
`            var myLabel = new Label();
            myLabel.FontFamily = (string)App.Current.Resources["GrialIconsLine"];
            myLabel.Text = GrialIconsFont.MapPin;
`

Please note that you also need to set the FontFamily.

All the code above requires `using UXDivers.Grial;`

Please take into consideration that I'm providing these snippets AS-IS, I didn't tested them but they should be work.

Hope this helps.

Bye.
-m

**Comment by Fatihpk** [comment]
Hi mrutter, thank you so much! All works great! 

Thanx for the good support ;)

**Comment by Fatihpk** [comment]
If i use the 'Navigation.PushModalAsync()' in your V3 the Modal shows up without a Close button on the right top corner. How do we get a Modal Page with a close button like your **Add Contact Page Flow** page. But then i want to call it again at the C# code :)

**Comment by mrutter** [comment]
Hi @Fatihpk,
I had the same doubt in the past.

In Grial UI Kit, the close button is introduced by the navigation page associated to the page.

The page **AddContactPage** is activated by the main dashboard thru its corresponding **ChatNavigationPage** that works as a container. **ChatNavigationPage** declares the toolbar's items (the close button). See SamplesCatalog.cs (line 211) and MainMenuViewModel.cs (line 41, line 61 ...).

A solution should be to use the same approach as Grial: create the navigation page that will wrap your page.

Bye.
-m

P.S. Please note that I'm not a Grial's employee; I'm a developer just like you, willing to help whenever I can. Also, I would like to suggest to you to open different tickets for different issues: that's would help other developers with the same needs.

**Comment by Fatihpk** [comment]
Just check't it, because i could not get it work in C# :(

**Comment by Fatihpk** [comment]
Thanx for helping me out even if you are not Grial's employee. Much appreciated. Do you have a sample code for the popup example?

**Comment by mrutter** [comment]
Hi @Fatihpk ,
you're welcome.

I created an empty page with a single button then I associated the Clicked event to the following code and it works fine:

`        private async void Button_Clicked(object sender, System.EventArgs e)
        {
            var page = new AddContactPage();
            var nav = new ChatNavigationPage(page);
            await Navigation.PushModalAsync(nav);
        }

[screen-recording.zip](https://github.com/UXDivers/Grial-UI-Kit-Support/files/3351038/screen-recording.zip)
`

Hope this helps.

Bye.
-m
